pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -151,6 +151,24 @@ def setup_environment(self) -> None:
         self.setup_distributed()
         super().setup_environment()
 
+    def setup(self, trainer: "pl.Trainer") -> None:
+        super().setup(trainer)
+        # share ddp pids to all processes
+        self._rank_0_has_called_call_children_scripts = self.broadcast(self._rank_0_has_called_call_children_scripts)
+        if self._should_run_deadlock_detection():
+            self._share_information_to_prevent_deadlock()
+
+        # move the model to the correct device
+        self.model_to_device()
+
+        if self.sync_batchnorm:
```

Comment:
side question for my knowledge. Why this is still in the codebase if it is not exposed as an argument to the plugin ?

Reply:
this is defined by the trainer: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L167